src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -169,14 +171,22 @@ public override void ExecuteCmdlet()
                                 env.Value?.GetEndpoint(AzureEnvironment.Endpoint.ResourceManager)?.ToLowerInvariant(),
                                 GeneralUtilities.EnsureTrailingSlash(ARMEndpoint)?.ToLowerInvariant(), StringComparison.CurrentCultureIgnoreCase));
 
-                        var newEnvironment = new AzureEnvironment { Name = this.Name };
+                        IAzureEnvironment newEnvironment = new AzureEnvironment { Name = this.Name };
+                        var defProfile = GetDefaultProfile();
+                        if (defProfile != null)
+                        {
+                            IAzureEnvironment _newEnvironment;
+                            defProfile.TryGetEnvironment(this.Name, out _newEnvironment);
```

Comment:
TryGetENvironment will return true if the environment is successfully created.  Also, do not use ```_``` as a prefix for local variables.  Instead the code could look like this:

```
IAzureEnvrironment newEnvironment
if (!defProfile.TryGetEnvironment(this.Name, out newEnvironment)
{
    new Environment = new AzureEnvironment {Name = this.Name};
}
```